### PR TITLE
integration: use dynamic client for CR checks

### PR DESCRIPTION
**Pull request by kopiczko**
Towards https://github.com/giantswarm/giantswarm/issues/4481.

This makes the check more generic. Next steps is to move it to `e2e-harness` and use it from there.

### CODE REVIEW

File: integration/setup/tenant_cluster.go [hunk 1]
```diff
@@ -87,14 +92,57 @@ func EnsureTenantClusterDeleted(ctx context.Context, id string, config Config, w
 	return nil
 }
 
-func crExistsCondition(ctx context.Context, id string, config Config) release.ConditionFunc {
+func crExistsCondition(ctx context.Context, config Config, crd *apiextensionsv1beta1.CustomResourceDefinition, crNamespace, crName string) release.ConditionFunc {
```
Comment (kopiczko) [inline integration/setup/tenant_cluster.go]: From signature you can see it's more generic. It needs only `rest.Config` to function. I'm going to move it to `e2e-harness` as a next steps.

File: integration/setup/tenant_cluster.go [hunk 2]
```diff
@@ -87,14 +92,57 @@ func EnsureTenantClusterDeleted(ctx context.Context, id string, config Config, w
...
+		{
+			var err error
+
+			c := config.Host.RestConfig()
+			configShallowCopy := *c
+			configShallowCopy.APIPath = "/apis"
+			configShallowCopy.GroupVersion = gv
+			if configShallowCopy.UserAgent == "" {
+				configShallowCopy.UserAgent = rest.DefaultKubernetesUserAgent()
+			}
```
Comment (kopiczko) [inline integration/setup/tenant_cluster.go]: I expect that to be a little bit less cryptic in `client-go@10.12.x` but I don't know yet.

**Comment by kopiczko** [comment]
> I think we may want to have a TODO issue but up to you.

You mean for `client-go@1.12.x` migration? Uncertain. We'll hit that and then comments will come in handy. Do you think todo issue can help here somehow? I can still add it while moving the code to `e2e-harness`.

**Comment by xh3b4sd** [comment]
I am not sure how the migration of the lib is supposed to go right now but if it is independent then we will just forget about your comments here, which is why I then would prefer to have a TODO issue.

**Comment by kopiczko** [comment]
This is a breaking change in the `client-go` so we can't forget.

**Pull request merged**